packages/server/src/tests/utilities/api/base.ts

```diff
@@ -136,11 +141,25 @@ export abstract class TestAPI {
             `Got an undefined expected value for header "${key}", if you want to check for the absence of a header, use headersNotPresent`
           )
         }
-        request = request.expect(key, value as any)
+        req = req.expect(key, value as any)
       }
     }
 
-    return await request
+    try {
+      return await req
+    } catch (e: any) {
+      // We've found that occasionally the connection between supertest and the
+      // server supertest starts gets reset. Not sure why, but retrying it
+      // appears to work. I don't particularly like this, but it's better than
+      // flakiness.
+      if (e.code === "ECONNRESET") {
+        if (attempt > 2) {
+          throw e
+        }
+        return await this._requestRaw(method, url, opts, attempt + 1)
+      }
```

Comment:
This has driven me to the edge today. I was seeing several different types of spurious error when running locally:

1. `invalid localapi request` I tracked this down to be _something_ to do with Tailscale, given that this error string is only found [here](https://github.com/tailscale/tailscale/blob/4cbef205698b532791ccf881bc100f7c57f1d9cf/ipn/localapi/localapi.go#L196) when [looking it up on Google](https://www.google.com/search?q=%22invalid+localapi+request%22&sca_esv=ab02dd2696e7e4b9&source=hp&ei=jgIDZq2oMei-hbIPxKyQ2A0&iflsig=ANes7DEAAAAAZgMQnoLCaHfedW8iA1uk1-NSK1UKtjXr&ved=0ahUKEwitp57utpKFAxVoX0EAHUQWBNsQ4dUDCBc&uact=5&oq=%22invalid+localapi+request%22&gs_lp=Egdnd3Mtd2l6IhoiaW52YWxpZCBsb2NhbGFwaSByZXF1ZXN0IjIHECEYChigATIHECEYChigAUj9FlAAWNAUcAB4AJABAJgBgQGgAfgRqgEEMjAuNrgBA8gBAPgBAZgCGaAC7BHCAgsQABiABBixAxiDAcICDhAuGIAEGLEDGMcBGNEDwgIREC4YgAQYsQMYgwEYxwEY0QPCAg4QLhiABBiKBRixAxiDAcICDhAAGIAEGIoFGLEDGIMBwgILEC4YgAQYsQMYgwHCAg4QLhiDARixAxiABBiKBcICCBAuGIAEGLEDwgIOEC4YgAQYsQMYgwEY1ALCAggQABiABBixA8ICCxAuGIAEGMcBGK8BwgIFEAAYgATCAggQLhixAxiABMICCxAuGIAEGLEDGNQCwgINEAAYgAQYsQMYRhj5AcICCBAAGIkFGKIEwgIIEAAYgAQYogTCAgUQIRigAcICCRAhGAoYoAEYCpgDAJIHBDE3LjigB_mSAQ&sclient=gws-wiz). Uninstalling Tailscale appears to have solved that.
2. `WebSockets request was expected` this appears to be _something_ to do with having a debugger attached while running tests, and indeed I haven't been able to get it to happen if the debugger isn't connected. It's quite rare when the debugger is connected, though.
3. `socket hang up` for whatever reason, we appear to get an `ECONNRESET` from time to time between `supertest` and the HTTP server that `supertest` creates in order to server test requests. I wasn't able to figure out exactly what's up, but I did observe that retrying when this happens usually succeeds the 2nd time. So that's what's going on here.